internal/app/machined/pkg/controllers/network/ethernet_status.go

```diff
@@ -0,0 +1,184 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package network
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/mdlayher/ethtool"
+	"github.com/siderolabs/gen/xslices"
+	"github.com/siderolabs/go-pointer"
+	"go.uber.org/zap"
+
+	"github.com/siderolabs/talos/internal/app/machined/pkg/controllers/network/watch"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/controllers/runtime"
+	"github.com/siderolabs/talos/pkg/machinery/nethelpers"
+	"github.com/siderolabs/talos/pkg/machinery/resources/network"
+)
+
+// EthernetStatusController reports Ethernet link statuses.
+type EthernetStatusController struct{}
+
+// Name implements controller.Controller interface.
+func (ctrl *EthernetStatusController) Name() string {
+	return "network.EthernetStatusController"
+}
+
+// Inputs implements controller.Controller interface.
+func (ctrl *EthernetStatusController) Inputs() []controller.Input {
```

Comment:
can't we generate this based on ethernet spec inputs?

Reply:
sorry, I didn't get that, how to generate Inputs? 

this one watches Linux kernel and reports `ethtool` status, so it doesn't actually have any resource inputs.

Reply:
ah got it, I was wondering why this does not have the spec as an input